src/rust/engine/testutil/mock/src/execution_server.rs

```diff
@@ -143,7 +144,7 @@ impl MockResponder {
       .received_messages
       .lock()
       .unwrap()
-      .push((message.descriptor().name().to_string(), Box::new(message)));
+      .push((message.descriptor().name().to_string(), Box::new(message), SystemTime::now()));
```

Comment:
Is this the right place to record the timestamp when server receives the request?

Reply:
Looks good :)